internal/connect/wrapper.go

```diff
@@ -148,9 +161,21 @@ func (w Wrapper) Register(regcode, instanceDataFile string) error {
 		extraData["instance_data"] = string(data)
 	}
 
+	// Provide profile data for registration
+	// make sure ProfileCache empty for registration
+	util.DeleteProfileCache()
+	profileData, err := FetchSystemProfiles(arch, false)
+	extraData["data_profiles"] = profileData
+
 	// NOTE: we are not interested in the code. Hence, we don't save it
 	// anywhere.
+	util.Debug.Print("hwinfo: ", hwinfo)
+	util.Debug.Printf("extraData: %+v", extraData)
```

Comment:
I'd rather not have these debug prints.

Reply:
I added this because this data was not being provided when --debug was used, and it was the only way to validate what data gets sent on registration.